x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -690,6 +690,7 @@ private void runComputeOnDataNode(
             dataNodeRequestExecutor.start();
             // run the node-level reduction
             var externalSink = exchangeService.getSinkHandler(externalId);
+            task.addListener(() -> exchangeService.finishSinkHandler(externalId, new TaskCancelledException(task.getReasonCancelled())));
```

Comment:
Could you stick `@Nullable` on the `exception` argument to `finishSinkHandler` and add a note that it'll fire any errors into the sync - if it is still running. Or something like that.

Reply:
Added in https://github.com/elastic/elasticsearch/pull/106392/commits/3c018f7b6c818346c87fd8da94dc10717978fc61.